src/aead.rs

```diff
@@ -17,6 +17,12 @@ use crate::hacl::chacha20_poly1305;
 
 use libcrux_platform::{aes_ni_support, simd128_support, simd256_support};
 
+#[cfg(feature = "std")]
```

Comment:
The global way of handling this is usually to add something like
```rust
#[cfg(feature = "std")]
extern crate std;
```
and mapping std and alloc to the same name with a `use`.
Also, when possible, we should add an `alloc` feature at some point, to allow for full `no_std`.

But this here is fine.